### PR TITLE
Cut v2.1.0-rc.0

### PR DESCRIPTION
**Pull request by mrueg**
**What this PR does / why we need it**:
```
## v2.1.0-rc.0 / 2021-05-20
  
* [FEATURE] Add support for native TLS #1354
* [FEATURE] Add wildcard option to metric-labels-allowlist #1403
* [FEATURE] Add build info metric #1332
* [CHANGE] Add "uid" label to every pod metric #1304
* [CHANGE] Add resourceVersion to CronJob metrics #1447
* [CHANGE] Update go version and dependencies #1474
* [CHANGE] Bump client-go and friends to v0.21 (Kubernetes v1.21) #1463
* [CHANGE] Replace deprecated use of ioutil #1458
* [BUGFIX] Fix builder.Builder WithMetrics signature #1455
* [BUGFIX] Fix pod-metric missing reasons #1287
* [BUGFIX] Fix multiListWatch resourceVersion mismatch if watch reconnected #1377
```

@lilic @tariq1890 @brancz

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,9 +68,9 @@ At most, 5 kube-state-metrics and 5 [kubernetes releases](https://github.com/kub
 
 | kube-state-metrics | **Kubernetes 1.17** |  **Kubernetes 1.18** |  **Kubernetes 1.19** |  **Kubernetes 1.20** |  **Kubernetes 1.21** |
 |--------------------|---------------------|----------------------|----------------------|----------------------|----------------------|
-| **v1.8.0**         |         -           |          -           |          -           |          -           |          -           |
 | **v1.9.8**         |         -           |          -           |          -           |          -           |          -           |
 | **v2.0.0**         |         -/✓         |          -/✓         |          ✓           |          ✓           |          -/✓         |
+| **v2.1.0-rc.0**    |         -/✓         |          -/✓         |          ✓           |          ✓           |          ✓           |
```
Comment (lilic) [inline README.md:73]: I don't think we need an rc, we can go straight to the release of v2.1, we mainly did this for 2.0 due to so many breaking changes. What do you think?

**Review by lilic — COMMENTED** [review]
Thank you!

File: README.md (hunk 1 above)
Comment (mrueg) [reply README.md:73]: This works for me as well, I did it as the release process suggested to start with an rc. I don't mind going straight to 2.1.0 and branch off then. the risk should be rather low.

Comment (tariq1890) [reply README.md:73]: I think we should still follow the documented release process.

But having said that, if we want to retire the rc builds, I am okay with amending the release process in another PR and going forward with this approach

**Review by lilic — APPROVED** [review]
As mentioned I am happy with going with releases candidates 👍 

> error: pathspec 'release-2.1' did not match any file(s) known to git

You need to create a new release branch, do you have permissions for it?

**Comment by mrueg** [comment]
> As mentioned I am happy with going with releases candidates +1
> 
> > error: pathspec 'release-2.1' did not match any file(s) known to git
> 
> You need to create a new release branch, do you have permissions for it?

I was following the release steps (to ensure they are correct). I can create a branch here for the release from master.

**Comment by mrueg** [comment]
I pushed the release-2.1 branch and switched this PR to point to it

**Pull request merged**